.pre-commit-config.yaml

```diff
@@ -25,6 +25,7 @@ repos:
     rev: v1.9.0
     hooks:
       - id: mypy
+        additional_dependencies: [types-python-dateutil]
```

Comment:
i think it's better to refactor `ResParser._parse_date` to use standard lib `datetime` functionality instead. `dateutil` was never added as a `pymatgen` dependency so the only reason this worked is because some other `pmg` dependency already uses `dateutil`

Reply:
Thanks! Wonderful suggestion. Sorry I wasn't thinking about the code itself much (as there were too many `mypy` errors).

Reply:
i definitely know the feeling. sometimes there are so many errors, you can't devote full attention to everyone. it's usually good to have a second look with fresh eyes in those cases 😄

Reply:
Yes. That's the reason I comment here and there so that me (and others) could have a closer look at those I'm unsure about. Thanks!